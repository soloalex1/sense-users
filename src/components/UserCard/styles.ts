import { GridListItem } from 'react-aria-components';
import styled from 'styled-components';

export const CardContainer = styled(GridListItem)`
  border: 2px solid black;
  padding: 8px;

  svg {
    height: 1.5rem;
    width: 1.5rem;
  }
`;

export const CardBody = styled.div`
  display: flex;
  flex-flow: column nowrap;
  gap: 4px;
`;
