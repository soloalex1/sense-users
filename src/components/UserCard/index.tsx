import { UserCircleIcon } from '@heroicons/react/24/outline';

import * as S from './styles';

import { User } from '../../types';

type UserCardProps = {
  user: User;
};

const UserCard = ({ user }: UserCardProps) => {
  return (
    <S.CardContainer id={user.id}>
      <UserCircleIcon />
      <S.CardBody>
        <h2>{user.name}</h2>
        <p>{user.phone}</p>
        <p>{user.email}</p>
      </S.CardBody>
    </S.CardContainer>
  );
};

export default UserCard;
