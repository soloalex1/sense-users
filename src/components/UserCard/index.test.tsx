import { render, screen } from '../../test/utils';

import { userMock } from '../../test/mockUsers';
import { User } from '../../types';

import UserCard from '.';
import { GridList } from 'react-aria-components';

describe('UserCard', () => {
  let randomUserIndex = 0;
  let randomUser = {} as User;

  const renderWithWrapper = (Component: JSX.Element) => {
    return render(<GridList aria-label="test grid list">{Component}</GridList>);
  };

  beforeEach(() => {
    randomUserIndex = Math.floor(Math.random() * userMock.length);
    randomUser = userMock[randomUserIndex];
  });

  it('should match the snapshot', () => {
    const user = userMock[0];

    const tree = renderWithWrapper(<UserCard user={user} />);
    expect(tree).toMatchSnapshot();
  });

  it('given a specific user, it should render its details correctly', () => {
    renderWithWrapper(<UserCard user={randomUser} />);

    const userName = screen.getByText(randomUser.name);
    expect(userName).toBeVisible();

    const userPhone = screen.getByText(randomUser.phone);
    expect(userPhone).toBeVisible();

    const userEmail = screen.getByText(randomUser.email);
    expect(userEmail).toBeVisible();
  });
});
