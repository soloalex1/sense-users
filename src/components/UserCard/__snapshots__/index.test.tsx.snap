// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserCard should match the snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <span
        data-focus-scope-start="true"
        hidden=""
      />
      <div
        class="react-aria-GridList"
        data-layout="stack"
        data-rac=""
        id="react-aria-:r0:"
        role="grid"
        tabindex="0"
      >
        <div
          aria-label="Leanne Graham"
          class="sc-blHHSb eIfQWY"
          data-key="1"
          data-rac=""
          id="react-aria-:r0:-1"
          role="row"
          tabindex="-1"
        >
          <div
            aria-colindex="1"
            role="gridcell"
            style="display: contents;"
          >
            <svg
              aria-hidden="true"
              data-slot="icon"
              fill="none"
              stroke="currentColor"
              stroke-width="1.5"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M17.982 18.725A7.488 7.488 0 0 0 12 15.75a7.488 7.488 0 0 0-5.982 2.975m11.963 0a9 9 0 1 0-11.963 0m11.963 0A8.966 8.966 0 0 1 12 21a8.966 8.966 0 0 1-5.982-2.275M15 9.75a3 3 0 1 1-6 0 3 3 0 0 1 6 0Z"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
            <div
              class="sc-gtLWhw gEHknJ"
            >
              <h2>
                Leanne Graham
              </h2>
              <p>
                1-770-736-8031 x56442
              </p>
              <p>
                Sincere@april.biz
              </p>
            </div>
          </div>
        </div>
      </div>
      <span
        data-focus-scope-end="true"
        hidden=""
      />
    </div>
  </body>,
  "container": <div>
    <span
      data-focus-scope-start="true"
      hidden=""
    />
    <div
      class="react-aria-GridList"
      data-layout="stack"
      data-rac=""
      id="react-aria-:r0:"
      role="grid"
      tabindex="0"
    >
      <div
        aria-label="Leanne Graham"
        class="sc-blHHSb eIfQWY"
        data-key="1"
        data-rac=""
        id="react-aria-:r0:-1"
        role="row"
        tabindex="-1"
      >
        <div
          aria-colindex="1"
          role="gridcell"
          style="display: contents;"
        >
          <svg
            aria-hidden="true"
            data-slot="icon"
            fill="none"
            stroke="currentColor"
            stroke-width="1.5"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M17.982 18.725A7.488 7.488 0 0 0 12 15.75a7.488 7.488 0 0 0-5.982 2.975m11.963 0a9 9 0 1 0-11.963 0m11.963 0A8.966 8.966 0 0 1 12 21a8.966 8.966 0 0 1-5.982-2.275M15 9.75a3 3 0 1 1-6 0 3 3 0 0 1 6 0Z"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
          <div
            class="sc-gtLWhw gEHknJ"
          >
            <h2>
              Leanne Graham
            </h2>
            <p>
              1-770-736-8031 x56442
            </p>
            <p>
              Sincere@april.biz
            </p>
          </div>
        </div>
      </div>
    </div>
    <span
      data-focus-scope-end="true"
      hidden=""
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
